salalib/nagent.cpp

```diff
@@ -187,7 +189,9 @@ void AgentSet::init(int agent, int trail_num)
 {
    if (m_release_locations.size()) {
       int which = pafrand() % m_release_locations.size();
-      at(agent).onInit( m_release_locations[which], trail_num );
+      std::set<int>::iterator it = m_release_locations.begin();

```

Comment:
Hmm, that is not great. If you need random access, std:: set is the wrong container.